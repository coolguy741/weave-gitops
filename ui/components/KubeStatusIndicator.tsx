import _ from "lodash";
import * as React from "react";
import styled from "styled-components";
import { Condition } from "../lib/api/core/types.pb";
import Flex from "./Flex";
import Icon, { IconType } from "./Icon";
import Text from "./Text";

type Props = {
  className?: string;
  conditions: Condition[];
  short?: boolean;
  suspended?: boolean;
};
export enum ReadyType {
  Ready = "Ready",
  NotReady = "Not Ready",
  Reconciling = "Reconciling",
}

export function computeReady(conditions: Condition[]): ReadyType {
  if (!conditions?.length) return undefined;
  const readyCondition =
    _.find(conditions, (c) => c.type === "Ready") ||
    _.find(conditions, (c) => c.type === "Available");
  if (readyCondition) {
    if (readyCondition.status === "True") return ReadyType.Ready;
    if (
      readyCondition.status === "Unknown" &&
      readyCondition.reason === "Progressing"
    )
      return ReadyType.Reconciling;
    return ReadyType.NotReady;
  }

  if (_.find(conditions, (c) => c.status === "False"))
    return ReadyType.NotReady;
  return ReadyType.Ready;
}

export function computeMessage(conditions: Condition[]) {
  if (!conditions?.length) return undefined;
  const readyCondition =
    _.find(conditions, (c) => c.type === "Ready") ||
    _.find(conditions, (c) => c.type === "Available");
  if (readyCondition) return readyCondition.message;

  const falseCondition = _.find(conditions, (c) => c.status === "False");
  if (falseCondition) return falseCondition.message;
  return conditions[0].message;
}

function KubeStatusIndicator({
  className,
  conditions,
  short,
  suspended,
}: Props) {
  let readyText;
  let icon;
  let iconColor;
  if (suspended) {
    readyText = "Suspended";
    icon = IconType.SuspendedIcon;
  } else {
    const ready = computeReady(conditions);
    if (ready === ReadyType.Reconciling) {
      readyText = ReadyType.Reconciling;
      icon = IconType.ReconcileIcon;
      iconColor = "primary";
    } else if (ready === ReadyType.Ready) {
      readyText = ReadyType.Ready;
      icon = IconType.CheckCircleIcon;
      iconColor = "success";
    } else {
      readyText = ReadyType.NotReady;
      icon = IconType.FailedIcon;
      iconColor = "alert";
    }
  }

  let text = computeMessage(conditions);
  if (short || suspended) text = readyText;

  return (
    <Flex start className={className} align>
      <Icon size="base" type={icon} color={iconColor} text={text} />
    </Flex>
  );
}

export default styled(KubeStatusIndicator).attrs({
  className: KubeStatusIndicator.name,
})`
  ${Icon} ${Text} {
    color: ${(props) => props.theme.colors.black};
    font-weight: 400;
  }
`;
