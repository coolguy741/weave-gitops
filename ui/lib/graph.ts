// A collection of helper functions to render a graph of kubernetes objects
// with in the context of their parent-child relationships.
import {
  Applications,
  GroupVersionKind,
  UnstructuredObject,
} from "./api/applications/applications.pb";

export type UnstructuredObjectWithParent = UnstructuredObject & {
  parentUid?: string;
};

// Kubernetes does not allow us to query children by parents.
// We keep a list of common parent-child relationships
// to look up children recursively.
export const PARENT_CHILD_LOOKUP = {
  Deployment: {
    group: "apps",
    version: "v1",
    kind: "Deployment",
    children: [
      {
        group: "apps",
        version: "v1",
        kind: "ReplicaSet",
        children: [{ version: "v1", kind: "Pod" }],
      },
    ],
  },
};

export const getChildrenRecursive = async (
  appsClient: typeof Applications,
  result: UnstructuredObjectWithParent[],
  object: UnstructuredObjectWithParent,
  lookup: any
) => {
  result.push(object);

  const k = lookup[object.groupVersionKind.kind];

  if (k && k.children) {
    for (let i = 0; i < k.children.length; i++) {
      const child: GroupVersionKind = k.children[i];

      const res = await appsClient.GetChildObjects({
        parentUid: object.uid,
        groupVersionKind: child,
      });

      for (let q = 0; q < res.objects.length; q++) {
        const c = res.objects[q];

        // Dive down one level and update the lookup accordingly.
        await getChildrenRecursive(
          appsClient,
          result,
          { ...c, parentUid: object.uid },
          {
            [child.kind]: child,
          }
        );
      }
    }
  }
};
