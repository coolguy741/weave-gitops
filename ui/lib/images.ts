/* eslint-disable */
// @ts-nocheck

import failedSrc from "data-url:../images/failed.svg";
import fancyCardBackgroundBlue from "data-url:../images/fancy-card-background-blue.svg";
import fancyCardBackgroundOrange from "data-url:../images/fancy-card-background-orange.svg";
import fancyCardBackground from "data-url:../images/fancy-card-background.svg";
import bg from "data-url:../images/General-backgroundok.png";
import logoSrc from "data-url:../images/logo.svg";
import signInWheel from "data-url:../images/SignInWheel.svg";
import successSrc from "data-url:../images/success.svg";
import suspendedSrc from "data-url:../images/suspended.svg";
import titleSrc from "data-url:../images/Title.svg";
import weaveLogo from "data-url:../images/WeaveLogo.svg";

export default {
  bg,
  failedSrc,
  fancyCardBackground,
  fancyCardBackgroundBlue,
  fancyCardBackgroundOrange,
  logoSrc,
  signInWheel,
  successSrc,
  suspendedSrc,
  titleSrc,
  weaveLogo,
};
