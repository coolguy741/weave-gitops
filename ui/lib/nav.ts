import qs from "query-string";
import { SourceRefSourceKind } from "./api/core/types.pb";
import { NoNamespace, PageRoute, V2Routes } from "./types";

// getParentNavValue returns the parent for a child page.
// This keeps the nav element highlighted if we are on a child page.
// Example: /sources and /git_repo will both show the "Sources" nav as selected.
export const getParentNavValue = (
  currentPage: any
): PageRoute | V2Routes | boolean => {
  switch (currentPage) {
    case V2Routes.Automations:
    case V2Routes.Kustomization:
    case V2Routes.HelmRelease:
      return V2Routes.Automations;

    case V2Routes.Sources:
    case V2Routes.GitRepo:
    case V2Routes.HelmChart:
    case V2Routes.HelmRepo:
    case V2Routes.Bucket:
      return V2Routes.Sources;

    case V2Routes.FluxRuntime:
      return V2Routes.FluxRuntime;

    default:
      // The "Tabs" component of material-ui wants a bool
      return false;
  }
};

const pageTitles = {
  [V2Routes.Automations]: "Applications",
  [V2Routes.Sources]: "Sources",
  [V2Routes.FluxRuntime]: "Flux Runtime",
};

export const getPageLabel = (route: V2Routes): string => {
  return pageTitles[route];
};

export const formatURL = (page: string, query: any = {}) => {
  return `${page}?${qs.stringify(query)}`;
};

export const formatSourceURL = (
  kind: SourceRefSourceKind,
  name: string,
  namespace: string = NoNamespace
) => {
  return formatURL(sourceTypeToRoute(kind), { name, namespace });
};

export function sourceTypeToRoute(t: SourceRefSourceKind): V2Routes {
  switch (t) {
    case SourceRefSourceKind.GitRepository:
      return V2Routes.GitRepo;

    case SourceRefSourceKind.Bucket:
      return V2Routes.Bucket;

    case SourceRefSourceKind.HelmRepository:
      return V2Routes.HelmRepo;

    case SourceRefSourceKind.HelmChart:
      return V2Routes.HelmChart;

    default:
      break;
  }
}
