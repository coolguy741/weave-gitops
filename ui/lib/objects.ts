import { stringify } from "yaml";
import { FluxObjectKind, Object as ResponseObject } from "./api/core/types.pb";

export enum Kind {
  GitRepository = "GitRepository",
  Bucket = "Bucket",
  HelmRepository = "HelmRepository",
  HelmChart = "HelmChart",
  Kustomization = "Kustomization",
  HelmRelease = "HelmRelease",
}

export function fluxObjectKindToKind(fok: FluxObjectKind): Kind {
  return Kind[FluxObjectKind[fok].slice(4)];
}

export class FluxObject {
  obj: any;

  constructor(response: ResponseObject) {
    this.obj = JSON.parse(response.payload);
  }

  yaml() {
    return stringify(this.obj);
  }

  // Return list of key-value pairs for the metadata annotations that follow
  // our spec
  metadata(): [string, string][] {
    const prefix = "metadata.weave.works/";
    const annotations = this.obj.metadata.annotations || {};
    return Object.keys(annotations).flatMap((key) => {
      if (!key.startsWith(prefix)) {
        return [];
      } else {
        return [[key.slice(prefix.length), annotations[key] as string]];
      }
    });
  }
}
