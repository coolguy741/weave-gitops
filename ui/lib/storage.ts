import { GitProvider } from "./api/applications/applications.pb";
import { PageRoute } from "./types";

const tokenKey = (providerName: GitProvider) =>
  `gitProviderToken_${providerName}`;

export function storeProviderToken(providerName: GitProvider, token: string) {
  if (!window.localStorage) {
    console.warn("no local storage found");
    return;
  }

  localStorage.setItem(tokenKey(providerName), token);
}

export function getProviderToken(providerName: GitProvider): string {
  if (!window.localStorage) {
    console.warn("no local storage found");
    return;
  }

  return localStorage.getItem(tokenKey(providerName));
}

const CALLBACK_STATE_KEY = "oauth_callback_state";

export type CallbackSessionState = { page: PageRoute; state: any };

export function storeCallbackState(value: CallbackSessionState) {
  if (!window.sessionStorage) {
    console.warn("no session storage found");
    return;
  }

  if (!value) {
    return null;
  }

  return sessionStorage.setItem(CALLBACK_STATE_KEY, JSON.stringify(value));
}

export function getCallbackState(): CallbackSessionState {
  const state = sessionStorage.getItem(CALLBACK_STATE_KEY);

  if (!state) {
    return null;
  }

  try {
    return JSON.parse(state);
  } catch (e) {
    return null;
  }
}

export function clearCallbackState() {
  return sessionStorage.removeItem(CALLBACK_STATE_KEY);
}
