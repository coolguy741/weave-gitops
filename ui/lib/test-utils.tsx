import { MuiThemeProvider } from "@material-ui/core";
import { createMemoryHistory } from "history";
import _ from "lodash";
import * as React from "react";
import { QueryClient, QueryClientProvider } from "react-query";
import { Router } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import AppContextProvider, { AppProps } from "../contexts/AppContext";
import { CoreClientContext } from "../contexts/CoreClientContext";
import {
  Applications,
  GetGithubAuthStatusRequest,
  GetGithubAuthStatusResponse,
  GetGithubDeviceCodeRequest,
  GetGithubDeviceCodeResponse,
  ParseRepoURLRequest,
  ParseRepoURLResponse,
  ValidateProviderTokenRequest,
  ValidateProviderTokenResponse,
} from "./api/applications/applications.pb";
import {
  Core,
  GetChildObjectsRequest,
  GetChildObjectsResponse,
  GetReconciledObjectsRequest,
  GetReconciledObjectsResponse,
  GetVersionRequest,
  GetVersionResponse,
  ListObjectsRequest,
  ListObjectsResponse,
} from "./api/core/core.pb";
import theme, { muiTheme } from "./theme";
import { RequestError } from "./types";

export type CoreOverrides = {
  GetChildObjects?: (req: GetChildObjectsRequest) => GetChildObjectsResponse;
  GetReconciledObjects?: (
    req: GetReconciledObjectsRequest
  ) => GetReconciledObjectsResponse;
  GetVersion?: (req: GetVersionRequest) => GetVersionResponse;
  ListObjects?: (req: ListObjectsRequest) => ListObjectsResponse;
};

export const createCoreMockClient = (
  ovr: CoreOverrides,
  error?: RequestError
): typeof Core => {
  const promisified = _.reduce(
    ovr,
    (result, handlerFn, method) => {
      result[method] = (req) => {
        if (error) {
          return new Promise((_, reject) => reject(error));
        }
        return new Promise((accept) => accept(handlerFn(req) as any));
      };

      return result;
    },
    {}
  );

  return promisified as typeof Core;
};

export type ApplicationOverrides = {
  GetGithubDeviceCode?: (
    req: GetGithubDeviceCodeRequest
  ) => GetGithubDeviceCodeResponse;
  GetGithubAuthStatus?: (
    req: GetGithubAuthStatusRequest
  ) => GetGithubAuthStatusResponse;
  ParseRepoURL?: (req: ParseRepoURLRequest) => ParseRepoURLResponse;
  ValidateProviderToken?: (
    req: ValidateProviderTokenRequest
  ) => ValidateProviderTokenResponse;
};

// Don't make the user wire up all the promise stuff to be interface-compliant
export const createMockClient = (
  ovr: ApplicationOverrides,
  error?: RequestError
): typeof Applications => {
  const promisified = _.reduce(
    ovr,
    (result, handlerFn, method) => {
      result[method] = (req) => {
        if (error) {
          return new Promise((_, reject) => reject(error));
        }
        return new Promise((accept) => accept(handlerFn(req) as any));
      };

      return result;
    },
    {}
  );

  return promisified as typeof Applications;
};

export function withTheme(element) {
  return (
    <MuiThemeProvider theme={muiTheme}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MuiThemeProvider>
  );
}

type TestContextProps = AppProps & { api?: typeof Core };

export function withContext(
  TestComponent,
  url: string,
  { api, ...appProps }: TestContextProps
) {
  const history = createMemoryHistory();
  history.push(url);
  const queryClient = new QueryClient({
    defaultOptions: { queries: { retry: false } },
  });
  const isElement = React.isValidElement(TestComponent);
  return (
    <Router history={history}>
      <AppContextProvider renderFooter {...appProps}>
        <QueryClientProvider client={queryClient}>
          <CoreClientContext.Provider value={{ api, featureFlags: {} }}>
            {isElement ? TestComponent : <TestComponent />}
          </CoreClientContext.Provider>
        </QueryClientProvider>
      </AppContextProvider>
    </Router>
  );
}
