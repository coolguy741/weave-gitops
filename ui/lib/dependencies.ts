import { Automation } from "../hooks/automations";
import { FluxObjectNode } from "./objects";

// findNode searches for a node by the node's name and namespace.
export function findNode(
  nodes: FluxObjectNode[],
  name: string,
  namespace: string
): FluxObjectNode | null {
  const matches = nodes.filter(
    (node) => node.name === name && node.namespace === namespace
  );

  if (matches.length > 0) {
    return matches[0];
  } else {
    return null;
  }
}

// getNeighborNodes returns nodes which depend on the current node
// or are dependencies of the current node.
export function getNeighborNodes(
  nodes: FluxObjectNode[],
  currentNode: FluxObjectNode
): FluxObjectNode[] {
  const dependencyNodes = currentNode.dependsOn
    .map((dependency) => {
      const name = dependency.name;
      let namespace = dependency.namespace;

      if (!namespace) {
        namespace = currentNode.namespace;
      }

      return findNode(nodes, name, namespace);
    })
    .filter((n) => n);

  const dependentNodes = nodes.filter((node) => {
    let isDependent = false;

    for (const dependency of node.dependsOn) {
      const name = dependency.name;
      let namespace = dependency.namespace;
      if (!namespace) {
        namespace = node.namespace;
      }

      if (name === currentNode.name && namespace === currentNode.namespace) {
        isDependent = true;
        break;
      }
    }

    return isDependent;
  });

  return dependencyNodes.concat(dependentNodes);
}

// getGraphNodes returns all nodes in the current node's dependency tree, including the current node.
export function getGraphNodes(
  nodes: FluxObjectNode[],
  automation: Automation
): FluxObjectNode[] {
  // Find node, corresponding to the automation.
  const currentNode = findNode(nodes, automation.name, automation.namespace);

  if (!currentNode) {
    return [];
  }

  currentNode.isCurrentNode = true;

  // Find nodes in the current node's dependency tree.
  let graphNodes: FluxObjectNode[] = [];

  const visitedNodes = new Map<string, boolean>();

  let nodesToExplore: FluxObjectNode[] = [currentNode].concat(
    getNeighborNodes(nodes, currentNode)
  );

  if (nodesToExplore.length === 1) {
    return [];
  }

  while (nodesToExplore.length > 0) {
    const node = nodesToExplore.shift();

    const newNodes = getNeighborNodes(nodes, node).filter(
      (n) => !visitedNodes[n.id]
    );

    for (const n of newNodes) {
      visitedNodes[n.id] = true;
    }

    nodesToExplore = nodesToExplore.concat(newNodes);

    graphNodes = graphNodes.concat(node);
  }

  // Remove duplicates from graphNodes.
  graphNodes = graphNodes.filter(
    (node, index) => graphNodes.indexOf(node) === index
  );

  return graphNodes;
}
