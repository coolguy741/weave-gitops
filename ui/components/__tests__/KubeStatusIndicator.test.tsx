import { render, screen } from "@testing-library/react";
import "jest-styled-components";
import React from "react";
import renderer from "react-test-renderer";
import { withTheme } from "../../lib/test-utils";
import KubeStatusIndicator from "../KubeStatusIndicator";

describe("KubeStatusIndicator", () => {
  it("renders ready", () => {
    const conditions = [
      {
        type: "Ready",
        status: "True",
        reason: "ReconciliationSucceeded",
        message:
          "Applied revision: main/a3a54ef4a87f8963b14915639f032aa6ec1b8161",
        timestamp: "2022-03-03 17:00:38 +0000 UTC",
      },
    ];
    render(withTheme(<KubeStatusIndicator conditions={conditions} />));

    const msg = screen.getByText(conditions[0].message);
    expect(msg).toBeTruthy();
  });
  it("renders when available", () => {
    const conditions = [
      {
        type: "Available",
        status: "True",
        reason: "ReconciliationSucceeded",
        message:
          "Applied revision: main/a3a54ef4a87f8963b14915639f032aa6ec1b8161",
        timestamp: "2022-03-03 17:00:38 +0000 UTC",
      },
    ];
    render(withTheme(<KubeStatusIndicator conditions={conditions} />));

    const msg = screen.getByText(conditions[0].message);
    expect(msg).toBeTruthy();
  });
  it("renders ready - short", () => {
    const conditions = [
      {
        type: "Ready",
        status: "True",
        reason: "ReconciliationSucceeded",
        message:
          "Applied revision: main/a3a54ef4a87f8963b14915639f032aa6ec1b8161",
        timestamp: "2022-03-03 17:00:38 +0000 UTC",
      },
    ];
    render(withTheme(<KubeStatusIndicator short conditions={conditions} />));

    const ready = screen.getByText("Ready");
    expect(ready).toBeTruthy();
  });
  it("renders an error", () => {
    const conditions = [
      {
        type: "Ready",
        status: "False",
        reason: "BigTrouble",
        message: "There was a problem",
        timestamp: "2022-03-03 17:00:38 +0000 UTC",
      },
    ];
    render(withTheme(<KubeStatusIndicator conditions={conditions} />));

    const msg = screen.getByText(conditions[0].message);
    expect(msg).toBeTruthy();
  });
  it("1593 - handles unhealthy", () => {
    const conditions = [
      {
        type: "Ready",
        status: "False",
        reason: "HealthCheckFailed",
        message:
          "Health check failed after 30.004470633s, timeout waiting for: [Deployment/test/backend status: 'Failed']",
        timestamp: "2022-03-03 16:55:29 +0000 UTC",
      },
      {
        type: "Healthy",
        status: "False",
        reason: "HealthCheckFailed",
        message: "HealthCheckFailed",
        timestamp: "2022-03-03 16:55:29 +0000 UTC",
      },
    ];

    render(withTheme(<KubeStatusIndicator conditions={conditions} />));

    const msg = screen.getByText(conditions[0].message);
    expect(msg).toBeTruthy();
  });
  describe("snapshots", () => {
    it("renders success", () => {
      const conditions = [
        {
          type: "Ready",
          status: "True",
          reason: "ReconciliationSucceeded",
          message:
            "Applied revision: main/a3a54ef4a87f8963b14915639f032aa6ec1b8161",
          timestamp: "2022-03-03 17:00:38 +0000 UTC",
        },
      ];
      const tree = renderer
        .create(withTheme(<KubeStatusIndicator conditions={conditions} />))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("renders error", () => {
      const conditions = [
        {
          type: "Ready",
          status: "False",
          reason: "BigTrouble",
          message: "There was a problem",
          timestamp: "2022-03-03 17:00:38 +0000 UTC",
        },
      ];
      const tree = renderer
        .create(withTheme(<KubeStatusIndicator conditions={conditions} />))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
