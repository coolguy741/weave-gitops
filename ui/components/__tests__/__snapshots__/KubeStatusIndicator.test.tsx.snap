// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KubeStatusIndicator snapshots renders error 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-family: 'proxima-nova',Helvetica,Arial,sans-serif;
  font-size: 14px;
  font-weight: 800;
  text-transform: none;
  font-style: normal;
}

.c4 svg {
  fill: !important;
  height: 16px;
  width: 16px;
}

.c4.downward {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c4.upward {
  -webkit-transform: initial;
  -ms-transform: initial;
  transform: initial;
}

.c4 .c5 {
  margin-left: 4px;
}

.c1 .c3 .c5 {
  color: #1a1a1a;
  font-weight: 400;
}

<div
  className="c0 c1 KubeStatusIndicator"
>
  <div
    className="c2 c3 c4"
  >
    <img
      src=""
    />
    <span
      className="c5 c6"
      size="medium"
    >
      There was a problem
    </span>
  </div>
</div>
`;

exports[`KubeStatusIndicator snapshots renders success 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-family: 'proxima-nova',Helvetica,Arial,sans-serif;
  font-size: 14px;
  font-weight: 800;
  text-transform: none;
  font-style: normal;
}

.c4 svg {
  fill: !important;
  height: 16px;
  width: 16px;
}

.c4.downward {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c4.upward {
  -webkit-transform: initial;
  -ms-transform: initial;
  transform: initial;
}

.c4 .c5 {
  margin-left: 4px;
}

.c1 .c3 .c5 {
  color: #1a1a1a;
  font-weight: 400;
}

<div
  className="c0 c1 KubeStatusIndicator"
>
  <div
    className="c2 c3 c4"
  >
    <img
      src=""
    />
    <span
      className="c5 c6"
      size="medium"
    >
      Applied revision: main/a3a54ef4a87f8963b14915639f032aa6ec1b8161
    </span>
  </div>
</div>
`;
